iowrappers/nearby_search.go

```diff
@@ -32,19 +36,38 @@ type PlaceSearchRequest struct {
 	BusinessStatus POI.BusinessStatus
 	// true if using precise geolocation instead of using a grander administrative area
 	UsePreciseLocation bool
+
+	PriceLevel POI.PriceLevel
 }
 
-func (c *MapsClient) GoogleMapsNearbySearchWrapper(ctx context.Context, location POI.Location, placeType string, radius uint, pageToken string) (resp maps.PlacesSearchResponse, err error) {
-	mapsReq := maps.NearbySearchRequest{
-		Type: maps.PlaceType(placeType),
+// Create NearbySearchRequest for maps NearbySearch, adjust key settings such as radius and price levels
+func CreateMapSearchRequest(reqIn *PlaceSearchRequest, placeType POI.LocationType, token string) (reqOut maps.NearbySearchRequest) {
+	// Adjust radius, minPrice and maxPrice settings in search request
+	var radius uint = reqIn.Radius
+	var minPrice, maxPrice maps.PriceLevel
+	if POI.IsPricyEatery(reqIn.PlaceCat, reqIn.PriceLevel) {
+		// increase search radius
+		radius = uint(math.Min(float64(reqIn.Radius*4), GoogleNearbySearchMaxRadiusMeters))
+
+		// set price filters to match priceLevel
+		minPrice, maxPrice = maps.PriceLevel(fmt.Sprint(reqIn.PriceLevel)), maps.PriceLevel(fmt.Sprint(reqIn.PriceLevel))
```

Comment:
I set both minPrice and maxPrice in the GoogleMap API's search request to match input PriceLevel, so that we want to scrape one particular "Pricy Eatery" locations. 

That being said, we can also remove the maxPrice field in the output API search request, as you suggested. We can scrape additional results. For example, the user requests for $$$, we also scrape $$$$ in the database.

Reply:
Let me know if removing `maxPrice` field is a better choice here.

Reply:
I think it is a good enhancement, we can enhance `PlaceSearchRequest` later to make the internal request more general.

Reply:
Just removed the maxPrice in the latest 1132956, so that one Map API search will look for restaurants with price >= users' input price.